### PR TITLE
[CC 1.0.0.20] Add Mutual SSL Authentication support for CC as an update

### PR DESCRIPTION
**Pull request by renuka-fernando**
## Purpose
$subject
Backporing: https://github.com/wso2/docs-apim/pull/6225/files

### Tasks
- [ ] Update the release date with the date the docker images are pushed

## Related Issue
https://github.com/wso2-enterprise/wso2-apim-internal/issues/2840

### Choreo Connect mTLS Doc

image

### APIM mTLS Doc

Existing page: https://apim.docs.wso2.com/en/4.0.0/design/api-security/api-authentication/secure-apis-using-mutual-ssl/

Updated with this PR (moved common in APIM and CC to "includes" location)

image

### CODE REVIEW

File: en/tools/config-catalog-generator-cc/data/enforcer/configs.json [hunk 1]
```diff
@@ -447,6 +447,47 @@
...
+          "name": "enforcer.security.mutualSSL",
+          "required": false,
+          "description": "Configurations related to Mutual SSL",
+          "params": [
+            {
+              "name": "certificateHeader",
+              "type": "string",
+              "required": true,
+              "default": "X-WSO2-CLIENT-CERTIFICATE",
+              "description": "Header name which client certificate coming from the downstream client"
```
Comment (ruthryi) [inline en/tools/config-catalog-generator-cc/data/enforcer/configs.json:463]: The header name from which the client certificate is coming is from the downstream client.

File: en/tools/config-catalog-generator-cc/data/router/configs.json [hunk 2]
```diff
@@ -349,6 +349,34 @@
       ],
       "exampleFile": "upstream_tls.toml"
     },
+    {
+      "title": "Downstream TLS",
+      "options": [
+        {
+          "name": "router.downstream.tls",
+          "required": false,
+          "description": "The configurations for SSL configuration related to the downstream in Choreo Connect.",
```
Comment (ruthryi) [inline en/tools/config-catalog-generator-cc/data/router/configs.json:358]: The configurations for SSL related to the downstream in Choreo Connect.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/mutual-ssl-authentication.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+     ``` tab="Format"
+      curl -k --location -X GET "<API_URL>" -H  "accept: application/json" -H  "Authorization: Bearer <access-token>" --key <client_private_key> --cert <client_public_certificate>
+     ```
+
+     ``` tab="Example"
+      curl -k --location -X GET 'https://localhost:9095/test/1.0/foo' -H 'accept: applicaition/json' -H 'Authorization: Bearer  0ee9aa70-d631-3401-b152-521b431036ca' --key privateKey.key --cert example.pem
+     ```
+
+!!!Note
```
Comment (ruthryi) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/mutual-ssl-authentication.md:74]: The note syntax seems to be wrong. Please check

File: en/tools/config-catalog-generator-cc/data/enforcer/configs.json (hunk 1 above)
Comment (renuka-fernando) [reply en/tools/config-catalog-generator-cc/data/enforcer/configs.json]: Fixed with https://github.com/wso2/docs-apim/pull/7065/commits/4ac3b72b3278ab8265e5f4635499d5ea37db0c04

File: en/tools/config-catalog-generator-cc/data/router/configs.json (hunk 2 above)
Comment (renuka-fernando) [reply en/tools/config-catalog-generator-cc/data/router/configs.json:358]: Fixed with https://github.com/wso2/docs-apim/pull/7065/commits/4ac3b72b3278ab8265e5f4635499d5ea37db0c04

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/mutual-ssl-authentication.md (hunk 3 above)
Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/mutual-ssl-authentication.md]: Fixed with https://github.com/wso2/docs-apim/pull/7065/commits/4ac3b72b3278ab8265e5f4635499d5ea37db0c04

**Review by ruthryi — APPROVED** [review]

**Pull request merged**